web/src/providers/Span/Span.provider.tsx

```diff
@@ -59,17 +60,27 @@ const SpanProvider = ({children}: IProps) => {
   const matchedSpans = useSelector(SpanSelectors.selectMatchedSpans);
   const searchText = useSelector(SpanSelectors.selectSearchText);
 
+  useEffect(() => {
```

Comment:
Doing some cleanup of `affectedSpans`, `selectedSpan` when the page is unmounted.